extensions/sql-database-projects/src/models/deploy/azureSqlClient.ts

```diff
@@ -0,0 +1,202 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SubscriptionClient, Subscription, Location } from '@azure/arm-subscriptions';
+import { SqlManagementClient, Server } from '@azure/arm-sql';
+import * as coreAuth from '@azure/core-auth';
+import { ResourceManagementClient, ResourceGroup } from '@azure/arm-resources';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import * as utils from '../../common/utils';
+import { IAccount, Tenant, Token, IAzureAccountService } from 'vscode-mssql';
+export interface AzureAccountSession {
+	subscription: Subscription,
+	tenantId: string,
+	account: IAccount,
+	token: Token
+}
+
+/**
+ * TokenCredential wrapper to only return the given token
+ */
+export class SQLTokenCredential implements coreAuth.TokenCredential {
+
+	constructor(private _token: Token) {
+	}
+
+	public getToken(_: string | string[], __?: coreAuth.GetTokenOptions): Promise<coreAuth.AccessToken | null> {
+		return Promise.resolve({
+			token: this._token.token,
+			expiresOnTimestamp: this._token.expiresOn || 0
+		});
+	}
+}
+
+/**
+ * Client module to call Azure APIs for getting or creating resources
```

Comment:
It might be nice to add these apis and the ability to create a new Azure server in the vscode mssql extension, in case another extension, like sql bindings, or the vscode mssql extension itself could benefit from being able to easily create these resources.